internal/driver/controllerserver_helper.go

```diff
@@ -403,25 +431,38 @@ func (cs *ControllerServer) validateCreateVolumeRequest(ctx context.Context, req
 // prepareVolumeParams prepares the volume parameters for creation.
 // It extracts the capacity range from the request, calculates the size,
 // and generates a normalized volume name. Returns the volume name and size in GB.
-func (cs *ControllerServer) prepareVolumeParams(ctx context.Context, req *csi.CreateVolumeRequest) (volumeName string, targetSizeGB int, size int64, err error) {
+func (cs *ControllerServer) prepareVolumeParams(ctx context.Context, req *csi.CreateVolumeRequest) (volumeName string, targetSizeGB int, size int64, encryptionStatus string, err error) {
 	log := logger.GetLogger(ctx)
 	log.V(4).Info("Entering prepareVolumeParams()", "req", req)
 	defer log.V(4).Info("Exiting prepareVolumeParams()")
 
+	// by default encryption is disabled
+	encryptionStatus = "disabled"
 	// Retrieve the capacity range from the request to determine the size limits for the volume.
 	capRange := req.GetCapacityRange()
 	// Get the requested size in bytes, handling any potential errors.
 	size, err = getRequestCapacitySize(capRange)
 	if err != nil {
-		return "", 0, 0, err
+		return "", 0, 0, "", err
 	}
 
 	preKey := linodevolumes.CreateLinodeVolumeKey(0, req.GetName())
 	volumeName = preKey.GetNormalizedLabelWithPrefix(cs.driver.volumeLabelPrefix)
 	targetSizeGB = bytesToGB(size)
+	// Check if encryption should be enabled
+	if req.GetParameters()[VolumeEncryption] == True {
+		supported, err := cs.isEncryptionSupported(ctx, cs.metadata.Region)
+		if err != nil {
+			return volumeName, targetSizeGB, size, encryptionStatus, err
+		}
+		if !supported {
+			return volumeName, targetSizeGB, size, encryptionStatus, errInternal("Volume encryption is not supported in the %s region", cs.metadata.Region)
```

Comment:
We have multi region support in the CSI driver so we shouldn't just use `cs.metadata.Region` here. Try to extract the region from the req. See `createLinodeVolume()` on how that is being done. 

Tbh, we should be extracting and setting the region here in this func rather than the `createLinodeVolume()`. `createLinodeVolume()` should simply take the region as parameter.

Reply:
I have refactored the functions for this change.